### PR TITLE
feat(insights): multiple breakdowns in trends

### LINKED ISSUE #938: Support multiple properties in 'Break down by'
**Is your feature request related to a problem? Please describe.**
I cannot break down browser usage by browser and browser version at the same time. I can filter by a browser and then breakdown by version, but I cannot get a list of all browsers + versions in one table.

**Describe the solution you'd like**
Being able to select multiple properties from the list such as "Browser" + "Browser Version".

**Describe alternatives you've considered**

**Additional context**

***Thank*** you for your feature request - we love each and every one!

### LINKED ISSUE #23156: ActorsQuery doesn't support numeric ranges coming from a breakdownFilter with bins
## Bug description

ActorsQuery in insights should filter actors by a numeric range that users can select in the breakdown filters. Currently, it just omits the `breakdown` from the query, so actors aren't filtered.

## How to reproduce

Example query:
```json
{
  "query": {
    "kind": "ActorsQuery",
    "source": {
      "kind": "InsightActorsQuery",
      "source": {
        "kind": "TrendsQuery",
        "series": [
          {
            "kind": "EventsNode",
            "event": "user signed up",
            "name": "user signed up",
            "math": "total"
          }
        ],
        "interval": "day",
        "trendsFilter": {
          "display": "ActionsLineGraph"
        },
        "breakdownFilter": {
          "breakdown_type": "person",
          "breakdown": "dataset_count",
          "breakdown_histogram_bin_count": 10
        },
        "properties": {
          "type": "AND",
          "values": [
            {
              "type": "AND",
              "values": [
                {
                  "key": "dataset_count",
                  "value": "1",
                  "operator": "gt",
                  "type": "event"
                }
              ]
            }
          ]
        }
      },
      "day": "2024-06-18T00:00:00-07:00",
      "series": 0,
      "breakdown": "[2.9,3.2]",
      "includeRecordings": true
    },
    "select": [
      "actor",
      "created_at",
      "event_count",
      "matched_recordings"
    ],
    "orderBy": [
      "event_count DESC, actor_id DESC"
    ],
    "search": "",
    "limit": 100,
    "offset": 0
  },
  "refresh": "async"
}
```

### Debug info

```shell
- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]
```

### PR DESCRIPTION
**Pull request by skoob13**
## Problem

This PR adds multiple breakdowns support to `InsightsQuery`, `ActorsQuery`, and `InsightActorsQueryOptions`.

Closes #938 and #23156

## Breakdown by before changes

single_breakdown

### Demo

multiple_breakdowns

### Actors queries

multiple_breakdowns_actors

### Breakdown options

multiple_breakdowns_options

### CODE REVIEW

**Comment by aspicer** [comment]
Wow @skoob13 this is amazing! I can't believe this is basically your second PR? What a starter project.

Started to take a cursory look and play around with it. One (minor) thing I noticed is that it doesn't respect breakdown limit:
image

**Comment by skoob13** [comment]
> One (minor) thing I noticed is that it doesn't respect breakdown limit with multiple numerics

@aspicer nice catch! I fixed it.

File: frontend/src/scenes/insights/filters/BreakdownFilter/GlobalBreakdownOptionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    className="min-h-10 px-2"
+                    checked={!breakdownHideOtherAggregation}
+                    onChange={() => setBreakdownHideOtherAggregation(!breakdownHideOtherAggregation)}
+                    label={
+                        <div className="flex gap-1">
+                            <span>Group remaining values under "Other"</span>
+                            <Tooltip
+                                title={
+                                    <>
+                                        If you have over {breakdownLimit ?? 25} breakdown options, the smallest ones are
```
Comment (webjunkie) [inline frontend/src/scenes/insights/filters/BreakdownFilter/GlobalBreakdownOptionsMenu.tsx:33]: Magic number here and further below... can be moved to a variable.

File: posthog/hogql_queries/insights/trends/breakdown.py [hunk 2]
```diff
@@ -48,108 +61,280 @@ def __init__(
...
-        return self.enabled and self.query.breakdownFilter.breakdown_histogram_bin_count is not None
-
-    def column_expr(self) -> ast.Alias:
-        if self.is_histogram_breakdown:
-            return ast.Alias(alias="breakdown_value", expr=ast.Field(chain=self._properties_chain))
-        if self.query.breakdownFilter.breakdown_type == "cohort":
-            if self.modifiers.inCohortVia == InCohortVia.LEFTJOIN_CONJOINED:
-                return ast.Alias(
-                    alias="breakdown_value",
+        if self.enabled:
```
Comment (webjunkie) [inline posthog/hogql_queries/insights/trends/breakdown.py:78]: Readability: Can swap condition for early exit

**Review by webjunkie — APPROVED** [review]
It's a lot of changes, so I didn't dig too deep 😅 
But had an overall look around and it looks good 👍🏻  I checked the frontend for the general behavior and it's very well done, also the moving of certain options and such. Good work! If nobody else finds blockers, I agree with Sandy that we can roll this out and see if anything comes up in prod.

One issue I found is some wonkyness around bin counts, but this is not the fault of this PR, but rather already present bugs: see #23139. Maybe something to tackle next while at it.

**Pull request merged**